README.md

```diff
@@ -178,7 +178,83 @@ Please note that SendStartedStatement and SendCompletedStatement are both firing
 
 In the OnApplicationQuit call, I showed an example of how to call the publisher with a custom verb. OnApplicationQuit is a hook that will fire when the user quits the application. If you go ahead and attempt to close out of the player you'll see it fire off stuff again.
 
-There's also a way to make fully custom statements but I'm still trying to figure out the best way to make it clear on how to construct the statements from scratch and what gets included in them.
+
+### Statement Hooks
+
+Sometimes in your unity application you need to have access to a given statement for one reason or another. For this reason there's a way to hook into the statement sendoff from within the application. The following example shows how this is done.
+
+```csharp
+using System;
+using UnityEngine;
+using LRS.Domain;
+using System.Text.Json.Nodes;
+
+class Hook : MonoBehaviour
+{
+
+    // function that fires every time a statement is sent off.
+    private void OnStatementSent(JsonObject statement)
+    {
+        Debug.Log(statement["verb"]["id"]);
+    }
+
+    // Unity boilerplate for adding hooks to handlers
+    private void OnEnable()
+    {
+        Publisher.OnStatementSent += OnStatementSent;
+    }
+
+    // cleanup
+    private void OnDisable()
+    {
+        Publisher.OnStatementSent -= OnStatementSent;
+    }
+
+}
+```
+
+### Customize Statements
+
+If you need to customize the shape of a statement at sendoff, this can be accomplished by overloading the `SendStatement` function with an argument that contains a higher order function that describes the changes you want to make. The following code shows how this is done.
+
+```csharp
+using System;
+using UnityEngine;
+using LRS.Domain;
+using System.Text.Json.Nodes;
+
+class CustomizeStatements : MonoBehaviour
+{
+    // LRS Credentials
+    public string lrsUrl;
+    public string lrsKey;
+    public string lrsSecret;
+
+    // define the publisher
+    private Publisher publisher{get {return new Publisher(lrsUrl,lrsKey,lrsSecret);}}
```

Comment:
Is this also C#-specific syntax (specifically the `get {...}`)? Also I would break this up into multiple lines (unless that would go against C# code styling conventions).